optimizely/event_builder.py

```diff
@@ -172,19 +182,39 @@ def _get_attributes(self, attributes):
     if not attributes:
       return []
 
-    for attribute_key in attributes.keys():
+    for attribute_key in sorted(attributes.keys()):
       attribute_value = attributes.get(attribute_key)
       # Omit falsy attribute values
       if attribute_value:
-        attribute = self.config.get_attribute(attribute_key)
-        if attribute:
+        # Check for reserved attributes
+        reserved_attrs = [ReservedAttributes.BUCKETING_ID, ReservedAttributes.USER_AGENT]
+        if attribute_key in reserved_attrs:
           params.append({
-            self.EventParams.EVENT_ID: attribute.id,
+            'entity_id': attribute_key,
             'key': attribute_key,
             'type': self.EventParams.CUSTOM,
-            'value': attribute_value,
+            'value': attribute_value
           })
 
+        else:
+          attribute = self.config.get_attribute(attribute_key)
+          if attribute:
+            params.append({
+              'entity_id': attribute.id,
+              'key': attribute_key,
+              'type': self.EventParams.CUSTOM,
+              'value': attribute_value,
+            })
+
+    # Append Bot Filtering Attribute
+    attribute_key = ReservedAttributes.BOT_FILTERING
+    params.append({
```

Comment:
Please add condition which was placed in javascript code. If no botFiltering is given or old datafile is used, in that case it shouldn't send the value of botFiltering.
https://github.com/optimizely/javascript-sdk/pull/99/files#diff-1fbd9d0f02956a420aeef6959fc24caaR73